### ISSUE TITLE
Deserializing subclasses of abstact class

### ISSUE THREAD

**Issue by Ficus-x** [aaubry/YamlDotNet#731]
Hi! I have an abstract class and its sub classes. In list of that abstact class i add its subclasses and it serializes perfectly. But when i want to deserialize it, it throws error `No node deserializer was able to deserialize the node into...`. Can i somehow avoid it?
My deserializer looks like:
```public static IDeserializer Deserializer { get; set; } = new DeserializerBuilder()
            .WithTypeConverter(new VectorsConverter())
            .WithTypeConverter(new ColorConverter())
            .WithTypeConverter(new AttachmentIdentifiersConverter())
            .WithNamingConvention(UnderscoredNamingConvention.Instance)
            .WithNodeDeserializer(inner => new ValidatingNodeDeserializer(inner), deserializer => deserializer.InsteadOf<ObjectNodeDeserializer>())
            .IgnoreFields()
            .IgnoreUnmatchedProperties()
            .Build();

**Comment by edwardcookemacu** [comment]
Let me rephrase this so I can make sure I understand your question you have. You have an abstract class, `Foo`. With 2 implementing classes, `Bar` and `Baz`. You want YamlDotNet to automatically deserialize into `Bar` or `Baz`? In the example classes bellow you would be trying to deserialize Main.

I suspect it's throwing an error when trying to instantiate the abstract class, since that is what is probably the object type of your property.


```csharp
public abstract class Foo {
 public string X
}
public string Main {
 public Foo Prop { get; set; }
}

public class Bar : Foo {
 public string Prop1 { get; set; }
}

public class Baz : Foo {
 public string Prop2 { get; set; }
}
```

**Comment by edwardcookemacu** [comment]
If that is the case, then the solution in this issue may help you.

https://github.com/aaubry/YamlDotNet/issues/605#issuecomment-817424757

**Comment by EdwardCooke** [comment]
No response for a few months, closing the issue.
